Dockerfile

```diff
@@ -0,0 +1,49 @@
+FROM nvidia/cuda:11.8.0-devel-ubuntu22.04 AS dev
+
+RUN apt-get update -y \
+    && apt-get install -y python3-pip python3-venv
+
+WORKDIR /vllm
+COPY requirements.txt requirements.txt
+RUN --mount=type=cache,target=/root/.cache/pip \
+    pip install -r requirements.txt
+    
+COPY requirements-dev.txt requirements-dev.txt
+RUN --mount=type=cache,target=/root/.cache/pip \
+    pip install -r requirements-dev.txt
+
+FROM dev AS build
+
+ARG max_jobs=4
+
+COPY csrc csrc
+COPY vllm vllm
+COPY pyproject.toml pyproject.toml
+COPY README.md README.md
+COPY MANIFEST.in MANIFEST.in
```

Comment:
Ideally here we will copy only csrc folder and build only the c++ code. If you copy vllm folder any changes to the python code causes slow rebuild of the c++ code, most of the time this is not needed.

Also the README.md file is kind of required during the build of the c++ code, but we can copy empty README.me while build the c++ code.

Reply:
Makes sense. Do you think it would be possible to build the extensions separate from the wheel, so that the wheel building step only bundles everything without having to rebuild?

Reply:
I'm not sure how to build the wheel with already build extensions, but it might be possible. I don't think we need to build the wheel. If you want we can do another container path for building the wheel if you want this to be used to publish the wheel files to pip?

Reply:
Ok I just saw you did another stage to build the wheel. This is ok I think.

Reply:
I think for now no need to build the wheel in the dockerfile, can do that later if decide to consolidate docker/CI